packages/@aws-cdk/cx-api/lib/cxapi.ts

```diff
@@ -4,6 +4,24 @@
 
 import { Environment } from './environment';
 
+/**
+ * Bump this to the library version if and only if the CX protocol changes.
+ *
+ * We could also have used 1, 2, 3, ... here to indicate protocol versions, but
+ * those then still need to be mapped to software versions to be useful. So we
+ * might as well use the software version as protocol version and immediately
+ * generate a useful error message from this.
+ *
+ * Note the following:
+ *
+ * - The versions are not compared in a semver way, they are used as
```

Comment:
Why not, though? Your verification code uses `semver` already and I reckon a protocol-breaking change should derive in a version bump... So you could totally semver that.